common/test/acceptance/tests/lms/test_lms_edxnotes.py

```diff
@@ -536,6 +535,57 @@ def test_no_content(self):
             "You have not made any notes in this course yet. Other students in this course are using notes to:",
             notes_page_empty.no_content_text)
 
+    def test_notes_works_correctly_with_xss(self):
+        """
+        Scenario: Note text & tags should be HTML and JS escaped
+        Given I am enrolled in a course with notes enabled
+        When I visit the Notes page, with a Notes text & tag containing JS code
+        Then I should see them escaped.
```

Comment:
I need more time to review next week.  But I'm not getting all this.  It seems like HTML0escaping is happening later, so I'm not sure what this is testing, because the assertions aren't looking for anything escaped.  Are you just checking that  everything is properly JavaScript-escaped and the page doesn't break?  The final HTML-escaping needs to be in a jasmine and/or Bokchoy test I imagine.

But, I'm in a rush, so I could be wrong. :)